### PR TITLE
boards: arm: add PAN1783A-PA evaluation board

### PR DESCRIPTION
**Pull request by pideu-sj**
The PAN1783A-PA evaluation board is a development tool for the nRF5340 from Nordic Semiconductor and the third evaluation board variant for the PAN1783 Module. The power amplifier (PA) version includes FEM support.

### CODE REVIEW

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst
Comment (nordicjm) [inline boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:2]: Seemingly the .jpg files seem to have grown in size, can you go back to the old images and just rename them instead of changing them?

Comment (pideu-sj) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:2]: Yes, I noticed, that the old ones have a different resolution. So I reworked all the images to the same resolution (width), which looks much better. What I could do is decrease the resolution again. What size should I not exceed for the images?

Comment (kartben) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:2]: Aiming for <50KB would be great. Please consider using webp format (which would also support transparency, assuming you have access to the "original" pictures).

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst
Comment (kartben) [inline boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:65]: these are HTTP 404 for me, can you check?

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst
Comment (pideu-sj) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:2]: I tried to find a good compromise between resolution and quality and changed them into webp format.

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst
Comment (pideu-sj) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst]: Yes, you are right. Have to wait for the team to finish the website in background. Have removed specific links and added one that leads to portfolio overview.

**Review by nordicjm — DISMISSED** [review]

**Review by nordicjm — COMMENTED** [review]
Would also be good to have your boards in MAINTAINERS file with maintainers for them

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783a_pa_evb_cpuapp.dts
Comment (anangl) [inline boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783a_pa_evb_cpuapp.dts:29]: This does not seem to be correct. I think you rather need something like:
```
	&gpio_fwd {
		status = "okay";
		fem {
			gpios = <&gpio0 19 0>, <&gpio0 22 0>;
		};
	};
```

Comment (pideu-sj) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783a_pa_evb_cpuapp.dts:29]: Ah okay, yeah because the only thing the appcore needs to know is that these two pins needs to be forwarded right?
I just did it this way because it was suggested in a Nordic Devzone Ticket that way:

> Is the overlay below already added to the application core? Can you try to modify and add it to child_image/remote_shell/board/nrf5340dk_nrf5340_cpuapp.overlay 
```
/ {
    nrf_radio_fem: fem_node {
        compatible = "skyworks,sky66407-11", "generic-fem-two-ctrl-pins";
        ctx-gpios = <&gpio0 19 GPIO_ACTIVE_HIGH>;
        crx-gpios = <&gpio0 21 GPIO_ACTIVE_HIGH>;
     };
};
```
from: https://devzone.nordicsemi.com/support/311726

But I will try your version and change it accordingly, if everything is still working. Thanks a lot!

Comment (anangl) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783a_pa_evb_cpuapp.dts:29]: > Ah okay, yeah because the only thing the appcore needs to know is that these two pins needs to be forwarded right?

Yes.

> But I will try your version and change it accordingly, if everything is still working. Thanks a lot!

You'll need to also add `/delete-node/ uart;` to what I suggested if you don't want UART pins to be forwarded by default.

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst
Comment (kartben) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:2]: Thanks! Nice to have transparent background too! Given the relatively resolution, you can actually aim for much smaller size without noticeable degradation in quality. `cwebp pan1783a_evb.webp -q 70 -o test.webp` gives me a 24K file. You may want to work with your original lossless file as the input of cwebp to get even better quality.

Sorry if that sounds nitpicky but given that the images live in the Git repo, adding only ~70K instead of 150K of binary files would actullay make quite a difference. In fact, given there's hardly any visible difference between the 3 PCB, may I suggest you only include *one* image as an illustration? Thanks!

**Review by anangl — DISMISSED** [review]

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst
Comment (pideu-sj) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:2]: Okay, I changed the description. So having just one image is ok for me now. With my original *.png as the source, I came down to a size of 41K for the webp image.

**Review by nordicjm — DISMISSED** [review]

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783a_pa_evb_cpuapp.dts
Comment (pideu-sj) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783a_pa_evb_cpuapp.dts]: Thanks for pointing me to this! Haven't noticed, that this will overwrite my disabled status in the `pan1783a_pa_evb_cpuapp.dts`. I will change that of course.

**Review by anangl — DISMISSED** [review]

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst
Comment (kartben) [inline boards/arm/pan1783_pan1783a_pan1783apa_evb/doc/index.rst:1]: Sorry that I didn't catch this earlier but do you really want things to live in a `.../pan1783_pan1783a_pan1783apa_evb/...` folder? It feels like this wouldn't scale all that well as new revisions get added and folder needs renaming (like this very PR is actually demonstrating).
I would recommend sticking to a `pan1783` folder for all the variants of the pan1783* evaluation boards.

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783_pan1783a_pan1783apa_cpuapp_common-pinctrl.dtsi
Comment (kartben) [inline boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783_pan1783a_pan1783apa_cpuapp_common-pinctrl.dtsi:1]: these series of common files also feels like it's not ideally named, and `pan1783_cpuapp_common-pinctrl.dtsi` (and similar for other files) would be better and more future-proof?

**Review by kartben — COMMENTED** [review]
Sorry that I'm catching this late but really feels like the naming of the main board folder + common files is not ideal? I will defer to other reviewers to see if that's a blocker and will leave this as a comment (esp. as I might be coming in late).
I understand from the initial PR that board revisions wouldn't work in this case, but at least shortening the common path should be done if possible IMO.
/cc @anangl

File: boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783_pan1783a_pan1783apa_cpuapp_common-pinctrl.dtsi
Comment (pideu-sj) [reply boards/arm/pan1783_pan1783a_pan1783apa_evb/pan1783_pan1783a_pan1783apa_cpuapp_common-pinctrl.dtsi:1]: I've also thought a lot about this. On first time I came to the conclusion, that it looks strange, when all other panXXXX boards having an *_evb postfix and the pan1783 having not. But I think you're right. It looks better and cleaner that way. Have changed it.

**Review by kartben — DISMISSED** [review]
Really appreciate the extra effort! Looks great, thanks!

**Comment by pideu-sj** [comment]
@anangl & @nordicjm
can we merge after last changes? :-)

**Review by nordicjm — DISMISSED** [review]

**Comment by nordicjm** [comment]
@pideu-sj needs rebase

**Review by nordicjm — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Comment by kartben** [comment]
@pideu-sj thanks for stepping up as a maintainer btw! Feel free to claim your Zephyr Maintainer badge here: https://forms.gle/oCw9iAPLhUsHTapc8 :)

**Pull request merged**
